metals/src/main/scala/scala/meta/internal/metals/FileDecoderProvider.scala

```diff
@@ -0,0 +1,263 @@
+   * metalsDecodeFile:/somedir/someFile.scala?decoder=semanticdb&originalURI=file:///somedir/someFile.scala
+   * metalsDecodeFile:/somedir/someFile.scala.semanticdb?decoder=semanticdb&originalURI=file:///somedir/someFile.scala.semanticdb
+   *
+   * errors get logged and sent back as the text contents
+   */
+  def decodedFileContents(uriAsStr: String): Future[String] = {
+
+    val result = Try(URI.create(uriAsStr)) match {
+      case Success(uri) =>
+        if (uri.getScheme != "metalsDecodeFile")
+          Left("URL `$uri` has invalid scheme `${uri.getScheme}`")
+        else {
+          val paramsArr = uri.getQuery.split("&").map(_.split("=", 2))
+          if (paramsArr.exists(_.length != 2))
+            Left(s"URI `$uri` contains invalid query `${uri.getQuery}`")
+          else {
+            Try {
+              paramsArr
+                .map(f =>
+                  URLDecoder.decode(f(0), StandardCharsets.UTF_8) ->
+                    URLDecoder.decode(f(1), StandardCharsets.UTF_8)
+                )
+                .toMap
+            } match {
+              case Success(params) =>
+                if (!params.contains("decoder"))
+                  Left(
+                    s"URI `$uri` contains no `decoder` argument `${uri.getQuery}`"
+                  )
+                else {
+                  if (!params.contains("originalURI"))
+                    Left(
+                      s"URI `$uri` contains no `originalURI` argument `${uri.getQuery}`"
+                    )
+                  else {
+                    val originalURIAsStr = params("originalURI")
+                    Try {
+                      originalURIAsStr.toAbsolutePath
+                    } match {
+                      case Success(filePath) =>
+                        if (!filePath.exists)
+                          Left(s"URI `$uri` file doesn't exist `$filePath`")
+                        else
+                          Right(decodeFileContents(filePath, params))
+                      case Failure(_) =>
+                        Left(s"Bad URI path `$originalURIAsStr`")
+                    }
+                  }
+                }
+              case Failure(_) => Left(s"Bad query parameters `$paramsArr`")
+            }
+          }
+        }
+      case Failure(_) => Left(s"Bad URI `$uriAsStr`")
+    }
+    result match {
+      case Right(success) => success
+      case Left(e) =>
+        scribe.error(e)
+        Future.successful(e)
+    }
+  }
+
+  private def getBoolParam(
+      name: String,
+      params: Map[String, String]
+  ): Boolean = {
+    params.get(name).map(_.toBoolean).getOrElse(false)
+  }
+
+  private def decodeFileContents(
+      filePath: AbsolutePath,
+      params: Map[String, String]
+  ): Future[String] = {
+    params("decoder") match {
+      case "javap" =>
+        decodeJavap(filePath, getBoolParam("verbose", params))
```

Comment:
I'll remove non-verbose

Reply:
I'm not sure about removing the non-verbose option. I took a look at [jvm-bytecode-viewer](https://github.com/mnxn/vscode-jvm-bytecode-viewer) to see the verbose output and it's significantly larger than normal mode. If someone isn't master in bytecode reading it could be overwhelming at the beginning.
Maybe non-verbose options could be commands and in the file-explorer context menu will have only verbose ones?

Reply:
You might be right @kpodsiad ! I will leave it up to you @Arthurm1

Reply:
I'll add non-verbose back in but only as a "show active file javap" command.  The context menu is getting big.

Reply:
Actually it doesn't matter now we have a sub-menu so I'll keep both on the sub-menu